### PR TITLE
Stationrc fix

### PR DESCRIPTION
**Pull request by fschlueter**
The changes in the PR so far are minor and irrelevant but I have a question to the code which I will comment on inline (edit: the inline feature did not work as expected, see next comment)

### CODE REVIEW

**Comment by fschlueter** [comment]
In the function `def automatch_phab(self, lab, match=1)` you start looping over all labs in line 146 `for i in labs:`, however later in the for loop (line 166 and following) you use `lab` instead of `i` as argument to some functions, e.g., `self.set_tmon(lab, self.tmon['SSTout'])`. When running `bring_up.py`, `lab` has the value `31` which means that `labs` is set to the following value at the beginning of the function" `labs=range(0, 24)`. Does it make sense to call e.g. `self.set_tmon` with the argument `31`?

@tikarg @cozzyd

File: lab4_controller.py [hunk 1]
```diff
@@ -167,13 +167,13 @@ def automatch_phab(self, lab, match=1):
...
                     time.sleep(0.1)
                     width = self.scan_width(scanNum)
-                    self.logger.warning(f"Width now {width}")
+                    self.logger.warning(f"LAB{i:<2}: Width now {width}")
                     if width < 200 or width > 4000:
-                        self.logger.error("still not working, bailing")
+                        self.logger.error(f"LAB{i:<2}: Still not working, bailing")
                         err = True
                         continue
                     self.l4reg(lab, 8, 2700)
```
Comment (fschlueter) [inline lab4_controller.py]: What is this line `self.l4reg(lab, 8, 2700)` doing there? It looks like it changes the register only if the `width` is in the desired range (but also only after it failed the first if condition) ...

**Comment by fschlueter** [comment]
Maybe @ryankrebs016  can also say something

**Comment by barawn** [comment]
Passing 31 to l4reg (via set_tmon) just sets all of them automatically, so it could be switched or left as-is.

**Comment by fschlueter** [comment]
> Hi @barawn, thanks for your reply!
> the same is true for `self.l4reg(lab, 8, 2500)` ?

Okay, checking what `set_tmon` does (e.g. calling `l4reg`) this answers my question

**Comment by fschlueter** [comment]
@barawn so I should keep everything as it is? Its still confusing that in a loop over all labs/channels one would execute in the loop a command to "all" labs.

**Comment by ryankrebs016** [comment]
might be safer to kick one at a time instead of all at once? And line 179 should be changed to self.l4reg(lab, 8, self.defaults[8]) I think. the 2700 was hardcoded for 3.2GHz

**Comment by fschlueter** [comment]
> might be safer to kick one at a time instead of all at once? 

And it might avoid future confusion... @barawn is it okay to change those line to call those function on only one lab at a time or has the use of 31, i.e., all lab, a deeper reason?

**Comment by cozzyd** [comment]
Should the 2500 also be changed? Or is that still a reasonable value at 2.4 GHz?

**Comment by ryankrebs016** [comment]
it's probably okay

**Comment by barawn** [comment]
Yeah, you can change it to call just the individual LABs (e.g. set_tmon(blah, i) or whatever). Doesn't actually matter, since it takes the same amount of time to program all of them as to program one. But conceptually it can be confusing, so might as well change it.

**Comment by fschlueter** [comment]
Okay let me use this PR to fix those issues.

File: lab4_controller.py (hunk 1 above)
Comment (fschlueter) [reply lab4_controller.py]: @ryankrebs016 and @cozzyd does it for you make sense to "kick" the labs all with a different value at the end?

**Comment by barawn** [comment]
@ryankrebs016 The 2500 should be changed to defaults[8] - 200 or something.

The point of that line is that you can start up in a situation where the delay line's so far off that the DLL is basically broken because the edges can't propagate through. So you just slow things down a bit to let the DLL converge.

So it seems weird because you're changing the value, then restoring it to the same value, but that's the point, it's just a startup issue.

**Comment by ryankrebs016** [comment]
> The 2500 should be changed to defaults[8] - 200 or something.

2622 vs 2500 might functionally work the same. But yeah, defaults[8]-200 would be most correct.

**Comment by fschlueter** [comment]
> Yeah, you can change it to call just the individual LABs (e.g. set_tmon(blah, i) or whatever). Doesn't actually matter, since it takes the same amount of time to program all of them as to program one. But conceptually it can be confusing, so might as well change it.

Ah I think I understand now

**Comment by fschlueter** [comment]
> @ryankrebs016 The 2500 should be changed to defaults[8] - 200 or something.
> 
> The point of that line is that you can start up in a situation where the delay line's so far off that the DLL is basically broken because the edges can't propagate through. So you just slow things down a bit to let the DLL converge.
> 
> So it seems weird because you're changing the value, then restoring it to the same value, but that's the point, it's just a startup issue.

ah I think I understand now

**Comment by fschlueter** [comment]
@all, can you check the code? I also implemented a loop to repeat the "kick" with `defaults[8] - 200`. Do you think it is sensible or does not make sense?

**Comment by barawn** [comment]
Programming the register multiple times doesn't do anything (although it doesn't hurt) but the additional loops might give the DLL a bit more time to migrate to a more correct value.

As I said in a (much longer) email, though, it probably makes more sense to move a lot of these checks to a separate function which just verifies that the overall LAB4D delay line is working correctly.

**Pull request merged**